### DISCUSSION TITLE
ask for Storage Permission

### DISCUSSION THREAD

**Discussion by crxed9** [rodit/SnapMod#128] [Ideas & Feature Suggestions]
After trying SnapMod on a different phone, where every feature seems to work, except saving, I had to find out through a logcat that the permission to storage was denied...doh!

I would be nice if SnapMod asks for Storage Permission either on install or when trying its first save.

**Comment by RevealedSoulEven** [comment]
Bro. If you use snapmod then it's sure that you already have a rooted device with xposed installed.
And this clarifies that you are an advanced android user. So everyone have knowledge about it.

Secondly, when you're going to read a phrase then no one's gonna teacher you alphabets. Keeping in mind, I mean to say you already are a pro user as you know about reading logcat.

But yes your point is right too. It should ask for storage permission. But I think it can even use root or xposed and it won't need storage permission. And I guess that it uses xposed permissions to do that and storage permission is not at all required. But it is added in case that doesn't works. I'm not sure about it btw

**Reply by rodit** [reply]
It does require storage permissions if you do not use Android's download manager to download files (which seems to be required on older versions of Android).

**Reply by Magna-Graecia** [reply]
For what it's worth, I'm on an older version of Android (8.1.0 as you know), and I'm able to save with "use download manager" enabled or disabled. I leave it enabled because that's where it was by default and I've had no issues with it, and also it gives me a download arrow notification in the status bar, which I like. Just adding for info's sake.

**Comment by rodit** [comment]
I agree, I will add this soon.
